src/Compilers/CSharp/Test/Semantic/Semantics/AwaitExpressionTests.cs

```diff
@@ -20,6 +20,85 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     /// </summary>
     public class AwaitExpressionTests : CompilingTestBase
     {
+        [Fact]
+        public void TestAwaitInfoExtensionMethod()
+        {
+            var text =
+@"using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+static class App{
+    public static async Task Main(){
+        var x = new MyAwaitable();
+        x.SetValue(42);
+
+        Console.WriteLine(await x + ""!"");
+    }
+}
+
+struct MyAwaitable
+{
+    private ValueTask<int> task;
+    private TaskCompletionSource<int> source;
+
+    private TaskCompletionSource<int> Source
+    {
+        get
+        {
+            if (source == null)
+            {
+                source = new TaskCompletionSource<int>();
+                task = new ValueTask<int>(source.Task);
+            }
+            return source;
+        }
+    }
+    internal ValueTask<int> Task
+    {
+        get
+        {
+            _ = Source;
+            return task;
+        }
+    }
+
+    public void SetValue(int i)
+    {
+        Source.SetResult(i);
+    }
+}
+
+static class MyAwaitableExtension
+{
+    public static System.Runtime.CompilerServices.ValueTaskAwaiter<int> GetAwaiter(this MyAwaitable a)
+    {
+        return a.Task.GetAwaiter();
+    }
+}";
+
+            var csCompilation = CreateCompilation(text, targetFramework: TargetFramework.NetCoreAppAndCSharp);
+            var tree = csCompilation.SyntaxTrees.Single();
+
+            var model = csCompilation.GetSemanticModel(tree);
+            var awaitExpression = tree.GetRoot().DescendantNodes().OfType<AwaitExpressionSyntax>().First();
```

Comment:
>awaitExpression [](http://example.com/codeflow?start=16&length=15)

Please assert string representation of the located node. This will make the test more readable and will ensure we are testing what is intended.